### PR TITLE
[FAB-15724] Use in-proc client for migration tests 

### PR DESCRIPTION
**Pull request by stephyee**
#### Type of change
- Test update

#### Description

In migration int tests,
- add additional orderer after solo->etcdraft migration has completed
- use in-proc client instead of chaincode/peer

#### Related issues

https://jira.hyperledger.org/browse/FAB-15724

### CODE REVIEW

File: integration/raft/migration_test.go [hunk 1]
```diff
@@ -201,11 +189,11 @@ var _ = Describe("Kafka2RaftMigration", func() {
 
 			By("3) Verify: create new channel, executing transaction")
 			network.CreateAndJoinChannel(orderer, channel2)
-			nwo.EnableCapabilities(network, channel2, "Application", "V2_0", orderer, network.Peer("Org1", "peer0"), network.Peer("Org2", "peer0"))
-			nwo.DeployChaincode(network, channel2, orderer, chaincode, network.PeersWithChannel(channel2)...)
-			RunExpectQueryRetry(network, peer, channel2, 100)
-			RunExpectQueryInvokeQuery(network, orderer, peer, channel2, 100)
-			RunExpectQueryInvokeQuery(network, orderer, peer, channel2, 90)
+
+			env = CreateBroadcastEnvelope(network, peer, channel2, []byte("hola"))
```
Comment (yacovm) [inline integration/raft/migration_test.go]: Doing an [nwo.Broadcast ](https://github.com/hyperledger/fabric/blob/master/integration/nwo/orderer_client.go#L22-L51)doesn't mean much, as it doesn't check that the transaction was included into some block.

Perhaps:

1. Add usage of a [Deliver](https://github.com/hyperledger/fabric/blob/master/integration/nwo/orderer_client.go#L54-L88) call to ensure a new block with the expected seq num has been created
2. Extract this repetitive pattern (send transaction and ensure block is created) into a function

**Review by yacovm — DISMISSED** [review]
Looks good, just a comment that can be addressed in a later PR or in this PR

File: integration/raft/migration_test.go [hunk 2]
```diff
@@ -131,9 +118,6 @@ var _ = Describe("Kafka2RaftMigration", func() {
 			By("Create & join first channel, deploy and invoke chaincode")
 			network.CreateAndJoinChannel(orderer, channel1)
 			nwo.EnableCapabilities(network, channel1, "Application", "V2_0", orderer, network.Peer("Org1", "peer0"), network.Peer("Org2", "peer0"))
```
Comment (jyellick) [inline integration/raft/migration_test.go]: Application capabilities don't matter if peers/chaincode are not involved.

File: integration/raft/migration_test.go (hunk 1 above)
Comment (jyellick) [reply integration/raft/migration_test.go]: +1 Performing the Deliver check would improve the quality of this test dramatically.

File: integration/raft/migration_test.go [hunk 3]
```diff
@@ -757,9 +784,6 @@ var _ = Describe("Kafka2RaftMigration", func() {
...
```
Comment (jyellick) [inline integration/raft/migration_test.go]: Again here

File: integration/raft/migration_test.go [hunk 4]
```diff
@@ -131,9 +118,6 @@ var _ = Describe("Kafka2RaftMigration", func() {
 			By("Create & join first channel, deploy and invoke chaincode")
 			network.CreateAndJoinChannel(orderer, channel1)
```
Comment (jyellick) [inline integration/raft/migration_test.go]: No need to Join channel, creating is sufficient if no peers.

File: integration/raft/migration_test.go [hunk 5]
```diff
@@ -34,10 +35,9 @@ import (
...
-		testDir   string
-		client    *docker.Client
-		network   *nwo.Network
-		chaincode nwo.Chaincode
+		testDir string
+		client  *docker.Client
+		network *nwo.Network
 
 		process                ifrit.Process
 		peerProc, brokerProc   ifrit.Process
```
Comment (jyellick) [inline integration/raft/migration_test.go]: One of the primary goals in using the broadcast client directly is to eliminate the need for peers from this test, so I would expect peerProc to go away.

**Review by yacovm — APPROVED** [review]

**Review by mastersingh24 — APPROVED** [review]

**Pull request merged**